controller/share_manager_controller.go

```diff
@@ -1110,19 +1304,103 @@ func (c *ShareManagerController) createPodManifest(sm *longhorn.ShareManager, an
 	return podSpec
 }
 
-// isResponsibleFor in most controllers we only checks if the node of the current owner is down
-// but in the case where the node is unschedulable we want to transfer ownership,
-// since we will create sm pod on the sm.Status.OwnerID when the sm starts
+// isShareManagerPodStale checks the associated lease CR to see whether the current pod (if any)
+// has fallen behind on renewing the lease.  If there is any error finding out, we assume not stale.
+func (c *ShareManagerController) isShareManagerPodStale(sm *longhorn.ShareManager) (stale bool, holder string, err error) {
+	log := getLoggerForShareManager(c.logger, sm)
+
+	if enabled, _ := c.ds.GetSettingAsBool(types.SettingNameEnableShareManagerFastFailover); !enabled {
+		// stale is false, holder is empty, err is nil
+		return
+	}
+
+	leaseName := sm.Name
+	lease, err := c.ds.GetLeaseRO(leaseName)
+	if err != nil {
+		// Even NotFound would be odd.
+		err = errors.Wrapf(err, "failed to retrieve lease for %v", leaseName)
```

Comment:
Can we suppress this err when calling from `isResponsibleFor` function? When user create a RWX from UI, the volume is in detached state at the begging. There is no lease object yet. This logs is currently printed out repeatedly which might be misleading

Reply:
For sure, good catch.